src/components/UserCards.tsx

```diff
@@ -159,7 +160,7 @@ function search(users: MinUserAndProfile[], searchTerm: string) {
 
   return users.filter(u => match(u.user.name) || (u.profile && (
     match(u.profile.性别) ||
-    visibleUserProfileFields.some(fl => match(u.profile?.[fl.field])))));
+    visibleUserProfileFields.some(fl => match(u.profile?.[fl.field] as string)))));
```

Comment:
Do this, and update most frontend references of zUserProfile to zStringUserProfile. This will also address the type issue with the picture cropping PR.

```
/**
 * ... (document why we need this)
 */
export const zStringUserProfile = z.object({
  '性别': zStr,
  '英文别名': zStr,
   ...
});

export const zUserProfile = zStringUserProfile.merge(z.object({
  '点赞': z.number().optional(),
}));